File: Leetcoding-Actions/Explore-Monthly-Challenges/2020-12/04-theKthFactorOfN.java
class Solution {
    public int kthFactor(int n, int k) {
        /**
        * T: O(n) and S: O(1)
        */
        int factorCount = 0;
        int i = 1;
        while (i <= n) {
            if (n % i == 0) factorCount++;
            if (factorCount == k) break;
            i++;
        }
        return factorCount == k ? i : -1;
    }
}

File: Leetcoding-Actions/Explore-Monthly-Challenges/2020-12/04-theKthFactorOfN.py
class Solution:
    def kthFactor(self, n: int, k: int) -> int:
        '''
        T: O(n) and S: O(1)
        '''
        i = 1
        factorCount = 0
        while i <= n:
            if n % i == 0:
                factorCount += 1
            if factorCount == k:
                break
            i += 1
        
        return i if factorCount == k else -1
